test/window-event.test.ts

```diff
@@ -3,47 +3,81 @@ import { conn } from './connect';
+import { Application } from '../src/api/application/application';
 import { cleanOpenRuntimes } from './multi-runtime-utils';
 
 // tslint:disable-next-line:no-function-expression
 describe('Window.', function() {
+
+    let fin: Fin;
+    let app: Application;
+    let win: _Window;
+
+    const appConfigTemplate = {
+        name: 'adapter-test-app-win-events',
+        url: 'about:blank',
+        uuid: 'adapter-test-app-win-events',
+        autoShow: true,
+        saveWindowState: false,
+        accelerator: {
+            devtools: true
+        },
+        opacity: 1
+    };
+
     // tslint:disable-next-line
     this.timeout(30000);
-    describe('addEventListener()', () => {
-        let fin: Fin;
-
-        const appConfigTemplate = {
-            name: 'adapter-test-app',
-            url: 'about:blank',
-            uuid: 'adapter-test-app',
-            autoShow: true,
-            accelerator: {
-                devtools: true
-            }
-        };
-
-        before(async () => {
-            await cleanOpenRuntimes();
-            fin = await conn();
-        });
 
-        describe('"closed"', () => {
+    before(async () => {
+        await cleanOpenRuntimes();
+        fin = await conn();
+    });
+
+    beforeEach(async () => {
+        app = await fin.Application.create(appConfigTemplate);
+        await app.run();
+        win = await app.getWindow();
+    });
 
-            before(() => fin.Application.create(appConfigTemplate).then(app => app.run()));
+    afterEach(async() => {
+        await app.close();
+    });
 
-            it('called', () => {
-                const spy = sinon.spy();
+    describe('"closed"', () => {
 
-                return fin.Application.wrap({ uuid: 'adapter-test-app' }).then(app => app.getWindow()).then(win => {
+        it('called', () => {
+            const spy = sinon.spy();
 
-                    win.on('closed', spy);
+            win.on('closed', spy);
 
-                    return win.close()
-                        .then(() => delayPromise())
-                        .then(() => assert(spy.calledOnce));
-                });
-            });
+            return win.close()
+                .then(() => delayPromise())
+                .then(() => assert(spy.calledOnce));
         });
+    });
+
+    describe('"options-changed"', () => {
+
+        it('called with payload', () => {
+            const optionName = 'opacity';
+            const spy = sinon.spy();
+            win.on('options-changed', spy);
 
+            const optionToUpdate = {
+                [optionName]: 0
+            };
+
+            const expectedDiffObject = {
+                optionName,
+                oldVal: 1,
+                newVal: 0
+            };
+
+            return win.updateOptions(optionToUpdate)
+                .then(() => delayPromise())
+                .then(() => spy.args[0][0])
+                .then(payload => assert.deepEqual(payload.diff[0], expectedDiffObject));
```

Comment:
Doe this need to be updated to the new `diff` as an object shape?

Reply:
yeah, will push a fix in a bit.